### DISCUSSION TITLE
API: `GET v1/bills` Reporting wrong `next_expected_match`

### DISCUSSION THREAD

**Discussion by lvitti** [firefly-iii/firefly-iii#8972] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

The `v1/bills` GET call is returning a response with an incorrect value in the `next_expected_match` field. The bill date is June 23, and there is a payment on June 15, but it is still returning the `next_expected_match` as June 23.

Sample from demo page:
```
https://demo.firefly-iii.org/api/v1/bills/10?start=2024-06-01&end=2024-06-31

{
  "data": {
    "type": "bills",
    "id": "10",
    "attributes": {
      "created_at": "2024-06-15T16:10:20+02:00",
      "updated_at": "2024-06-15T16:12:03+02:00",
      "currency_id": "1",
      "currency_code": "EUR",
      "currency_symbol": "€",
      "currency_decimal_places": 2,
      "name": "My Bill",
      "amount_min": "100.00",
      "amount_max": "300.00",
      "date": "2024-06-23T00:00:00+02:00",
      "end_date": null,
      "extension_date": null,
      "repeat_freq": "monthly",
      "skip": 0,
      "active": true,
      "order": 1,
      "notes": null,
      "object_group_id": null,
      "object_group_order": null,
      "object_group_title": null,
      "next_expected_match": "2024-06-23T00:00:00+02:00",
      "next_expected_match_diff": "Expected 1 week from now",
      "pay_dates": [
        "2024-06-23T00:00:00+02:00"
      ],
      "paid_dates": [
        {
          "transaction_group_id": "452",
          "transaction_journal_id": "457",
          "date": "2024-06-15T00:00:00+02:00",
          "date_object": "2024-06-15T09:15:00.000000Z"
        }
      ]
    },
    "links": {
      "0": {
        "rel": "self",
        "uri": "/bills/10"
      },
      "self": "https://demo.firefly-iii.org/api/v1/bills/10"
    }
  }
}

```

### Debug information

demo site

### Expected behaviour

It must return null in that field because a payment has already been made in the period I'm filtering

### Steps to reproduce

1. Create a new Bill with a date later than the current one.
2. Save the Automation Rule.
3. Create a Expense that match the automation rule criteria.
4. Query the API

**Comment by JC5** [comment]
Not a bug. By setting the period to the 23rd, a bill is expected to be paid anywhere between May 23 and June 23. You paid on the 15th, so the next one is due June 23rd again.

**Reply by lvitti** [reply]
Oh, I'm sorry, I think I misunderstood the concept of `next_expected_match`.

So, to determine if the bill has already been paid, can I use `paid_dates`, right?
